app/src/test/java/edu/usf/cutr/otp/demo/chicago/PlanApiUnitTest.kt

```diff
@@ -0,0 +1,155 @@
+package edu.usf.cutr.otp.demo.chicago
+
+import edu.usf.cutr.otp.plan.api.PlanApi
+import edu.usf.cutr.otp.plan.model.Planner
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Assert
+import org.junit.Test
+import java.util.concurrent.CountDownLatch
+
+class PlanApiUnitTest {
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun testPlanApi() {
+        val planApi =
+            PlanApi("https://raw.githubusercontent.com/CUTR-at-USF/opentripplanner-client-library/tests/library/src/jvmTest/kotlin/edu/usf/cutr/otp/resources/chicago/plan.json")
+        var planner = Planner()
+        val latch = CountDownLatch(1)
+
+        planApi.getPlan(
+            success = {
+                latch.countDown()
```

Comment:
For all the API tests, I would do `latch.countDown()` after assigning the variable. Otherwise, there is the potential that this thread could be suspended after `latch.countDown()` but before the variable assignment executes, which could result in randomly failing tests when the main thread resumes with the asserts before the variables are assigned.

Reply:
Good find. Thank you.